src/main/java/com/gtnewhorizons/neid/mixins/early/minecraft/MixinBlock.java

```diff
@@ -0,0 +1,104 @@
+package com.gtnewhorizons.neid.mixins.early.minecraft;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import net.minecraft.block.Block;
+import net.minecraft.init.Blocks;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+
+/**
+ * This mixin exists to completely re-work the way that block harvestability is calculated. Forge adds this system
+ * on-top of vanilla, which creates a pre-populated array on each block with the size of the maximum metadata value. For
+ * instance, in vanilla we get a String array, and an int array both of size 16, pre-populated with null and -1,
+ * respectively. This is awful for scalability with adding more possible Blocks, and increasing the metadata to 16-bits
+ * from 4. This uproots the under-workings of that system to use a default and dynamic HashMaps for metadata specific
+ * values, so we only use the memory we need, because most things don't even take advantage of this system, especially
+ * based on block metadatas. The public methods of this system maintain API compatibility, so anything using those
+ * should still function perfectly the same. The only potential breakage is if something were to reflect/ASM/mixin with
+ * the private values/methods we're overwriting.
+ */
+@Mixin(Block.class)
+public class MixinBlock {
+
+    @Unique
+    private String neid$defaultHarvestTool = null;
+
+    @Unique
+    private int neid$defaultHarvestLevel = -1;
+
+    @Unique
+    private Map<Integer, String> harvestToolMap = new HashMap<Integer, String>();
```

Comment:
Perhaps consider depending on GTNHLib to use `Int2ObjectOpenHashMap` rather than boxed keys?

Reply:
Hmm, I don't have a problem with the dependency, there's probably enough stuff depending on GTNHLib or starting to that people won't hate needing it.

Reply:
This has been done.